web/lib/components/flashcards/Sets.jsx

```diff
@@ -1,28 +1,231 @@
+    (confirm = false) =>
+    async (e) => {
+      e.preventDefault()
+
+      confirm && (await doAction())
+      onOpenChange(false)
+    }
+
+  return (
+    <Dialog
+      title={_e('flashcardset.modal.confirm')}
+      open={open}
+      onOpenChange={onOpenChange}
+      dismissable={false}>
+      <div className="p-2">
+        <p className="mb-5 text-red-500">
+          {_e('flashcardset.modal.doYouWantToDeleteThisSet')}
         </p>
+        <div className="grid grid-cols-2 gap-2">
+          <Button
+            className="rounded border border-sky-300 bg-sky-100 py-2 text-sm font-semibold text-sky-700 hover:border-sky-400"
+            onClick={handleClick(true)}>
+            {_e('common.confirm')}
+          </Button>
+          <Button
+            className="rounded border border-slate-300 bg-slate-100 py-2 text-sm font-semibold text-slate-700 hover:border-slate-400"
+            onClick={handleClick()}>
+            {_e('common.cancel')}
+          </Button>
+        </div>
+      </div>
+    </Dialog>
+  )
+}
+
+function FlashcardSetDropdown({ set, updateSet, deleteSet }) {
+  const { _e } = useTranslation()
+  const [isEditDialogOpen, setIsEditDialogOpen] = useState(false)
+  const [isConfirmModalOpen, setIsConfirmModalOpen] = useState(false)
+  const { notify } = useContext(ToastContext)
+
+  const {
+    isOpen,
+    referenceProps,
+    doToggleDropdown,
+    doCloseDropdown,
+    floatingProps,
+  } = useDropdown({ placement: 'bottom-end' })
+
+  const toggleDropdown = (e) => {
+    e.preventDefault()
+    doToggleDropdown()
+  }
+
+  const openEditDialog = (e) => {
+    e.preventDefault()
+    setIsEditDialogOpen(true)
+  }
+
+  const doRemoveSet = async () => {
+    // @TODO: replace this with remove set action
+    await deleteSet.mutateAsync({ id: set.id })
+
+    if (deleteSet.status === 'error') {
+      notify({ title: _e('flashcardset.error.deleleSet'), variant: 'error' })
+    }
+
+    if (deleteSet.status === 'success') {
+      notify({
+        title: _e('flashcardset.deleteSetSuccessfully'),
+      })
+    }
+  }
+
+  const doUpdateSet = (name) => async (e) => {
+    e.preventDefault()
+
+    await updateSet.mutateAsync({ id: set.id, name: name, userId: set.user_id })
+    /* Cannot use `updateSet.status === 'success'` somehow.
+     * First time always return status as 'idle' which cause the
+     * dialog remain open.
+     */
+    if (updateSet.status !== 'error') {
+      notify({ title: _e('flashcardset.updateNameSuccessfully') })
+      setIsEditDialogOpen(false)
+    }
+  }
+
+  const openDeleteConfimDialog = (e) => {
+    e.preventDefault()
+
+    setIsConfirmModalOpen(true)
+  }
+
+  return (
+    <div className="relative">
+      <Button
+        className="rounded-full px-0 py-1 text-slate-400 hover:text-slate-600 hover:shadow-sm"
+        onClick={toggleDropdown}
+        onBlur={doCloseDropdown}
+        {...referenceProps()}>
+        <DotsVerticalIcon className="h-6 w-6" />
+      </Button>
```

Comment:
Maybe just use normal button here, this style isn't good.
<img width="59" alt="image" src="https://user-images.githubusercontent.com/172503/165141114-b3d4e677-ffe6-4f43-b110-767bceaa6046.png">